browser-interface/packages/shared/world/runtime-7/engine.ts

```diff
@@ -47,7 +61,30 @@ function getUserData(userId: string) {
   }
 }
 
+function getScene(id: string | number): SceneWorker | undefined {
+  if (typeof id === 'string') return getSceneWorkerBySceneID(id)
+  if (typeof id === 'number') return getSceneWorkerBySceneNumber(id)
+  return undefined
+}
+
+function getEngineFromScene(scene?: SceneWorker): IEngine | undefined {
+  return scene?.rpcContext.internalEngine?.engine
+}
+
 export const localProfileChanged = mitt<LocalProfileChange>()
+export const audioStreamEmitter = mitt<AudioStreamChange>()
+
+// AudioStream updates
+audioStreamEmitter.on('changeState', ({ entityId, sceneId, state }) => {
```

Comment:
if you move this listener inside the engine function, there is no need to expose the sceneWorker or the engine.

You have the engine there, and all the context of that scene lives in that function

Reply:
yeah, but we would be applying the new state to all the scenes for the same entity right?

Reply:
we can put the conditionals inside the function. Check this https://github.com/decentraland/unity-renderer/pull/6163/files#diff-1562784c24abb585b205a56650940d8c13d9740a3d19f1d1853039a4764a56b3R238

`if (sceneId !== scenId) {}`